code/entities/weapons/fnb/GooglyEye.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class GooglyEye : WeaponBase
+{
+    static private float MaxPlacementDistance = 96.0f;
+    PreviewEntity PreviewModel { get; set; }
+
+    public override void ClientSpawn()
+    {
+        base.ClientSpawn();
+
+        if (!PreviewModel.IsValid()) 
+        {
+            PreviewModel = new PreviewEntity
+            {
+                Model = Model.Load("models/googly_eyes/preview_googly_eyes_01.vmdl")
+            };
+        }
+    }
+    protected virtual bool IsPreviewTraceValid(TraceResult tr)
+    {
+        if (!tr.Hit || !tr.Entity.IsValid()) return false;
```

Comment:
Purely my preference, i would merge these two checks (line 24 and 26)

Reply:
Merged these into one